GYBITG/GalleryViewController.swift

```diff
@@ -1,15 +1,258 @@
-//
+//This class will take a video for the Gallery if the camera is available. Will have the app access the Photo Library if the camera is not available or if user selects video library from the actionsheet choices.
+//Will get the selected video and Save video to the main photo album.
+//Will conform to the UINavigationControllerDelegate, UIImagePickerControllerDelegate.
+//Will have a UIImageView to store a thumbnail of the video, a videoFileName,
+//Will convert the Date in the specified format "MMM dd,yyyy" - May 29,2019. Will convert the time to CMTime print the duration. Will Return an int for the number of rows in the table a row for each item. Will fill the tableView with the items in the repository array of Videos.
 //  GalleryViewController.swift
 //  GYBITG
 //
 //  Created by Student Account on 5/7/19.
 //
-
+import AVFoundation
+import Photos
 import UIKit
+import MobileCoreServices
+import CoreMedia
+
+//The protocol for the VideoView
+protocol VideoRepositoryProtocol{
+    //to store all videos
+    var videos: [Video] {get}
+    func createVideo(userID: String, videoURL: URL) -> Video
+    func getAllVideos() -> [Video]
+    func getVideo(videoID: String) -> Video?
+    func addVideo(videoToAdd: Video) -> String
+    func updateVideo(videoToUpdateID: String, description: String?, longerVideoURL: URL?, thumbnail: UIImage?) -> String?
+    func deleteVideo(videoToDeleteID: String) -> String?
+    func deleteAllVideos()
+    
+}
 
-class GalleryViewController: UIViewController {
+class GalleryViewController: UITableViewController, UINavigationControllerDelegate, UIImagePickerControllerDelegate {
+    
+    var videoRepository = MockVideoRepository()
```

Comment:
Creating the video repository here will make it harder to write unit tests for your viewcontroller. Instead, use your protocol to make your var of that type, without assigning it a value...
`var videoRepository: VideoRepositoryProtocol?`

Since your MockVideoRepository class conforms to the protocol
`class MockVideoRepository: VideoRepositoryProtocol`

in the AppDelegate, you can instantiate your MockVideoRepository and assign it to variable
`galleryViewController.videoRepository = MockVideoRepository()`
(Note you need a bit of code to get a reference to the galleryViewController - James is also working on this model, so check with him.)

Reply:
Thanks Brian, I have the code for the changes and will incorporate with the code James has, once he does his merge.